src/server/worker.cc

```diff
@@ -442,115 +439,146 @@ Status Worker::EnableWriteEvent(int fd) {
 }
 
 Status Worker::Reply(int fd, const std::string &reply) {
-  std::unique_lock<std::mutex> lock(conns_mu_);
-  auto iter = conns_.find(fd);
-  if (iter != conns_.end()) {
-    iter->second->SetLastInteraction();
-    redis::Reply(iter->second->Output(), reply);
+  if (ConnMap::accessor accessor; conns_.find(accessor, fd)) {
+    accessor->second->SetLastInteraction();
+    redis::Reply(accessor->second->Output(), reply);
     return Status::OK();
   }
 
   return {Status::NotOK, "connection doesn't exist"};
 }
 
 void Worker::BecomeMonitorConn(redis::Connection *conn) {
-  {
-    std::lock_guard<std::mutex> guard(conns_mu_);
-    conns_.erase(conn->GetFD());
-    monitor_conns_[conn->GetFD()] = conn;
+  if (ConnMap::accessor accessor; conns_.find(accessor, conn->GetFD())) {
```

Comment:
does this change the previous syntax? Previous if `conn->GetFD()` not erase anything, the insert would still happens?

Reply:
Hi @mapleFU ,

Thanks very much for your advice! 😊
Yes, this will change the behavior of `monitor_conns_`.
I have moved the `monitor_conns_` inserting out of the scope.

Best Regards,
Edward